azure-iot-device/samples/async-hub-scenarios/receive_message_x509.py

```diff
@@ -27,26 +27,25 @@ async def main():
         hostname=hostname, device_id=device_id, x509=x509
     )
 
-    # define behavior for receiving a message
-    async def message_listener(device_client):
-        while True:
-            message = await device_client.receive_message()  # blocking call
-            print("the data in the message received was ")
-            print(message.data)
-            print("custom properties are")
-            print(message.custom_properties)
+    # Define behavior for receiving a message
+    # NOTE: this could be a function or a coroutine
+    def message_received_handler(message):
+        print("the data in the message received was ")
+        print(message.data)
+        print("custom properties are")
+        print(message.custom_properties)
+
+    # Set the message received handler on the client
+    device_client.on_message_received = message_received_handler
```

Comment:
>device_client.on_message_received = message_received_handler [](start = 4, length = 60)

I notice that you almost always set handlers _after_ you call connect, and I know that this is significant because setting a handler can cause a connection to happen.  This sample doesn't call connect, yet it still works and that's confusing.  Since many developers copy samples and build off of them, I would recommend adding a comment to each handler assignment in the samples to say something like "If the client isn't connected, this assignment will cause the client to connect." or something like that.  (Have I mentioned that I don't like having an assignment causing a network connection?  Yes, I'm a broken record....)